### PR TITLE
CHARMS RAS -  Verifying fields in NV Participant Consent Record and Subject Flags.

### PR DESCRIPTION
**Pull request by katm007**
Verifying fields in NV Participant Consent Record and Subject Flags.

### CODE REVIEW

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java
Comment (sarwarahmed1) [inline src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:1]: There are a lot of if statements here that have no code in them, do they serve any purpose?

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java [hunk 1]
```diff
@@ -545,11 +547,167 @@ public void study_team_member_logs_in_to_Native_View_and_completes_consent_call(
...
             CommonUtils.selectDropDownValue(ras_NV_Consent_Record_TestDataManager.RESPONSE_TYPE, nativeViewCHARMSParticipantConsentPage.rasStudyConsentResponseTypeDropDown);
+            CommonUtils.sleep(500);
+        }
+        if (!ras_NV_Consent_Record_TestDataManager.CONSENT_CALL_SCHEDULED_TIME.isEmpty()) {
+            CucumberLogUtils.scenario.log("* * * * * CONSENT CALL SCHEDULED TIME * * * * *");
+            CommonUtils.clickOnElement(nativeViewCHARMSParticipantConsentPage.rasStudyConsentCallScheduleTimeCalendar);
+            CommonUtils.waitForVisibility(nativeViewCHARMSParticipantConsentPage.rasStudyConsentCallScheduleTimeTodayButton);
+            CommonUtils.clickOnElement(nativeViewCHARMSParticipantConsentPage.rasStudyConsentCallScheduleTimeTodayButton);
+            CommonUtils.clickOnElement(nativeViewCHARMSParticipantConsentPage.rasStudyConsentCallScheduleTimeAcceptButton);
+        }
```
Comment (sarwarahmed1) [inline src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:565]: Please fix the spacing between each if statement

**Review by sarwarahmed1 — CHANGES_REQUESTED** [review]
Please review comments

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java
Comment (katm007) [reply src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:1]: Yes, they're going to be filled eventually. This is for a method that will complete the consent call in NV for different participants.

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java (hunk 1 above)
Comment (katm007) [reply src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:565]: Do you mean that I should remove the spaces between the {} ?

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java
Comment (sarwarahmed1) [reply src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:1]: Got it, just wanted to make sure

File: src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java (hunk 1 above)
Comment (sarwarahmed1) [reply src/test/java/ServiceNow/CHARMS/Steps/RAS_All_Steps.java:564]: Yes

File: src/test/java/ServiceNow/CHARMS/Pages/NativeViewCHARMSSubjectFlagsPage.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    @FindBy(xpath = "//button[@id='sysverb_insert_and_stay']")
+    public WebElement subjectFlagsSaveButton;
+
+    /**
+     * Native View Subject Flags The following mandatory fields are not filled in: IBMFS Affected Status. Error Text
+     */
+    @FindBy(xpath = "//span[@class='outputmsg_text']")
+    public WebElement subjectFlagsIBMFSAffectedStatusNotFilledInErrorText;
+
```
[the remark this answers is not in the document]
Comment (katm007) [reply src/test/java/ServiceNow/CHARMS/Pages/NativeViewCHARMSSubjectFlagsPage.java:45]: Line removed

**Review by sarwarahmed1 — APPROVED** [review]

**Review by andreyeve — APPROVED** [review]

**Pull request merged**